lib/carrier/messaging/connection.ex

```diff
@@ -47,14 +50,85 @@ defmodule Carrier.Messaging.Connection do
 
   """
   # Again, this spec is what comes from emqttc
-  @spec connect(Keyword.t()) :: {:ok, connection()} | :ignore | {:error, term()}
-  def connect(opts) do
-    connect_timeout = Keyword.get(opts, :connect_timeout, @default_connect_timeout)
-
+  @spec start_link(pid(), Keyword.t()) :: {:ok, connection()} | :ignore | {:error, term()}
+  def start_link(owner, opts \\ []) do
     opts = opts
     |> add_connect_config
     |> add_internal_credentials
 
+    GenServer.start_link(__MODULE__, [owner, opts])
+  end
+
+  @doc """
+  Creates a GenMqtt-style reply endpoint and subscribes `subscriber` to the topic
+
+  `subscriber` defaults to the caller's pid
+  """
+  @spec create_reply_endpoint(conn::connection(), subscriber::pid()) :: {:ok, String.t} | {:error, atom()}
+  def create_reply_endpoint(conn, subscriber \\ self()) do
+    GenServer.call(conn, {:create_reply_endpoint, subscriber}, :infinity)
+  end
+
+  @doc """
+  Creates a subscription to a given topic for `subscriber`
+
+  `subscriber` defaults to caller's pid
+  """
+  @spec subscribe(conn::connection(), topic::String.t, subscriber::pid()) :: {:ok, String.t} | {:error, atom()}
+  def subscribe(conn, topic, subscriber \\ self()) do
+    GenServer.call(conn, {:subscribe, topic, subscriber}, :infinity)
+  end
+
+  @doc """
+  Removes the named subscription for `subscriber`.
+  Returns true if unsubscribe was successful, false otherwise.
+  """
+  @spec unsubscribe(conn::connection(), topic::String.t, subscriber::pid()) :: boolean()
+  def unsubscribe(conn, topic, subscriber \\ self()) do
+    GenServer.call(conn, {:unsubscribe, topic, subscriber}, :infinity)
+  end
+
+  @doc """
+  Publishes a message to a MQTT topic
+
+  ## Keyword Arguments
+
+    * `:routed_by` - the topic on which to publish `message`. Required.
+  """
+  @spec publish(conn::connection(), message::Map.t, opts::publish_opts()) :: :ok | {:error, atom()}
+  def publish(conn, message, opts) do
+    GenServer.call(conn, {:publish, message, opts}, :infinity)
+  end
+
+  @doc """
+  Sends a GenMqtt call message and waits for reply
+  """
+  @spec call(conn::connection(), topic::String.t, endpoint::String.t, message::Map.t, opts::call_opts()) :: Map.t | {:error, atom()}
+  def call(conn, topic, endpoint, message, opts \\ []) do
+    subscriber = Keyword.get(opts, :subscriber, self())
+    timeout = Keyword.get(opts, :timeout, 5000)
+    GenServer.call(conn, {:call, topic, endpoint, message, subscriber, timeout}, :infinity)
+  end
+
+  @doc """
+  Sends a GenMqtt cast message
+  """
+  @spec cast(conn::connection(), topic::String.t, endpoint::String.t, message::Map.t) :: :ok | {:error, atom()}
+  def cast(conn, topic, endpoint, message) do
+    GenServer.call(conn, {:cast, topic, endpoint, message}, :infinity)
```

Comment:
Sounds good... this was just the first time I noticed infinity = 1 hour.

Some infinities are larger than other infinities, though, so it's cool 👅